app/views/bookings/index.html.erb

```diff
@@ -0,0 +1,18 @@
+<h1>Calendar</h1>
+
+<%= month_calendar do |date| %>
+    <div id="<%=date%>">
+        <%= date %>
+        <% @laundries.each do |room| %>
```

Comment:
Since there will only be ONE Laundry, do we really need to query the database for a collection?

Reply:
#### Cucumber fails when we try to change that line

We tried to
* change index method in bookings_controller to have 
```
def index
  @laundry = Laundry.first
end
 ```
* update the view with @laundry
```
...
<% @laundry.schedule.occurrences_between(date.midnight, date.midnight + 24.hours).each do |slot| %>
...
```
When we do this, we get the following error message when running cucumber:
```
Given I am on the "Landing" page # features/step_definitions/basic_steps.rb:1
      undefined method `schedule' for nil:NilClass (ActionView::Template::Error)
      ./app/views/bookings/index.html.erb:6:in `block in _app_views_bookings_index_html_erb__2567584605677730085_47267011198920'
      ./app/views/bookings/index.html.erb:3:in `_app_views_bookings_index_html_erb__2567584605677730085_47267011198920'
      ./features/step_definitions/basic_steps.rb:2:in `"I am on the {string} page"'
      features/user_can_see_calendar.feature:8:in `Given I am on the "Landing" page'
```

Reply:
If it’s not a collection, then you need to change the view and remove the iteration

Reply:
yes, I took away line 6

Reply:
Here is my code in index.html.erb
```
<%= month_calendar do |date| %>
    <div id="<%=date%>">
        <%= date %>
            <% @laundry.schedule.occurrences_between(date.midnight, date.midnight + 24.hours).each do |slot| %>
                <div>
                    <%  if @laundry.check_availability(time: slot.time, amount: 1) && user_signed_in? %>
                        <%= link_to slot.to_formatted_s(:time), bookings_path(slot: slot), method: "post", data: {confirm: "Would you like to book this timeslot?"} %>
                    <% else %>
                        <p> <%= slot.to_formatted_s(:time) %></p>
                    <% end %>
                </div>
            <% end %>
    </div>
<% end %>
```

Reply:
When I run pry schedule is not an issue:
```
[1] pry(#<#<Class:0x0000556c1ceb2cc8>>)> @laundry
/home/laura/.rvm/gems/ruby-2.4.3/gems/ice_cube_chosko-0.2.0/lib/ice_cube/validations/hour_of_day.rb:8: warning: constant ::Fixnum is deprecated
=> #<Laundry:0x0000556c1cf42828
 id: 49,
 schedule:
  #<IceCube::Schedule:0x0000556c1d55d910
   @all_exception_rules=[],
   @all_recurrence_rules=
    [#<IceCube::DailyRule:0x0000556c1cef1900
      @interval=1,
      @uses=0,
      @validations=
       {:interval=>[#<IceCube::Validations::DailyInterval::Validation:0x0000556c1cef1130 @interval=1>],
        :base_min=>[#<IceCube::Validations::ScheduleLock::Validation:0x0000556c1cec8168 @type=:min>],
        :base_sec=>[#<IceCube::Validations::ScheduleLock::Validation:0x0000556c1ceb0658 @type=:sec>],
        :hour_of_day=>
         [#<IceCube::Validations::HourOfDay::Validation:0x0000556c1ce22ee8 @hour=8>,
          #<IceCube::Validations::HourOfDay::Validation:0x0000556c1ce22dd0 @hour=11>,
          #<IceCube::Validations::HourOfDay::Validation:0x0000556c1ce22880 @hour=14>,
          #<IceCube::Validations::HourOfDay::Validation:0x0000556c1ce22830 @hour=17>]}>],
   @start_time=Sat, 02 Mar 2019 00:00:00 UTC +00:00>,
 created_at: Sat, 02 Mar 2019 12:01:56 UTC +00:00,
 updated_at: Sat, 02 Mar 2019 12:01:56 UTC +00:00,
 capacity: 1>
[2] pry(#<#<Class:0x0000556c1ceb2cc8>>)> Laundry.first == @laundry
=> true
[3] pry(#<#<Class:0x0000556c1ceb2cc8>>)> @laundry.schedule
=> #<IceCube::Schedule:0x0000556c1d55d910
 @all_exception_rules=[],
 @all_recurrence_rules=
  [#<IceCube::DailyRule:0x0000556c1cef1900
    @interval=1,
    @uses=0,
    @validations=
     {:interval=>[#<IceCube::Validations::DailyInterval::Validation:0x0000556c1cef1130 @interval=1>],
      :base_min=>[#<IceCube::Validations::ScheduleLock::Validation:0x0000556c1cec8168 @type=:min>],
      :base_sec=>[#<IceCube::Validations::ScheduleLock::Validation:0x0000556c1ceb0658 @type=:sec>],
      :hour_of_day=>
       [#<IceCube::Validations::HourOfDay::Validation:0x0000556c1ce22ee8 @hour=8>,
        #<IceCube::Validations::HourOfDay::Validation:0x0000556c1ce22dd0 @hour=11>,
        #<IceCube::Validations::HourOfDay::Validation:0x0000556c1ce22880 @hour=14>,
        #<IceCube::Validations::HourOfDay::Validation:0x0000556c1ce22830 @hour=17>]}>],
 @start_time=Sat, 02 Mar 2019 00:00:00 UTC +00:00>
```

Reply:
But when I run cucumber I get the error 
```
undefined method `schedule' for nil:NilClass
```
and I can't understand why.

Reply:
Solved! We were missing a background in the old feature file! All green now.